templates/terraform/decoders/kms_key_ring.go.erb

```diff
@@ -0,0 +1,24 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2018 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+	config := meta.(*Config)
+	res["self_link"] = d.Get("name").(string)
+	d.SetId(res["name"].(string))
+	if err := parseImportId([]string{"projects/(?P<project>[^/]+)/locations/(?P<location>[^/]+)/keyRings/(?P<name>[^/]+)"}, d, config); err != nil {
+        	return nil, err
+	}
+	res["project"] = d.Get("project").(string)
```

Comment:
Instead of these, I'd recommend putting `ignore_read` on these attributes in terraform.yaml- that should make it so their d.Set calls don't appear.